great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DomainBuilderConfig(DictDot):
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during DomainBuilderConfig instantiation', k, v
+            )
```

Comment:
@donaldheppner dataclasses (both mutable and frozen) don't play nicely with **kwargs, which we need to maintain forwards compatibility. 

The existing pattern works well but I'd be more than happy to brainstorm a better solution.

Reply:
@cdkini Any interest in using the `f-string` format?..

Reply:
I believe f-strings always create the string while %s-substitution only creates the string if the associated log level is set. It's a minor optimization but can be relevant when considering the size of our codebase: https://docs.python.org/3/howto/logging.html#optimization

I won't die on this hill if you feel strongly about f-strings through 😄

Reply:
alternatively we could just store this as a dictionary (which should play nicely with dataclasses). I'd like to understand this usecase

Reply:
@cdkini You are right.  Glad to see the explanation!  Thanks!